compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1967,6 +1975,17 @@ splitConArgs_maybe con args = do
 -- work. Constructor workers are not handled (yet).
 convertDataCon :: Env -> GHC.Module -> DataCon -> [LArg Var] -> ConvertM (LF.Expr, [LArg Var])
 convertDataCon env m con args
+    | AllowLargeTuples False <- envAllowLargeTuples env
+    , NameIn GHC_Tuple fsName <- con
+    , let name = unpackFS fsName
+    -- TODO: A better way to detect uses of tuples than this string malarkey
+    , let (front, middle, end) = (head name, init (tail name), last name)
```

Comment:
Is there any way of doing this more nicely with a new/existing pattern synonym?

Reply:
Unfortunately, the data constructors for tuples are just basic data constructors at this stage, so I can't disambiguate them easily except through checking the strings and the point of import

Reply:
I've defined a pattern synonym which at least cleans up the code in LFConversion and offloads it to UtilGHC